os/linux_base.h

```diff
@@ -123,12 +138,67 @@ typedef void (*dispatch_mach_msg_destructor_t)(void*);
 #endif
 #define OS_NOTHROW
 
+#ifdef OS_ASSUME_NONNULL_BEGIN
+#undef OS_ASSUME_NONNULL_BEGIN
+#endif
+#define OS_ASSUME_NONNULL_BEGIN
+
+#ifdef OS_ASSUME_NONNULL_END
+#undef OS_ASSUME_NONNULL_END
+#endif
+#define OS_ASSUME_NONNULL_END
+
+#ifdef OS_SWIFT_UNAVAILABLE
+#undef OS_SWIFT_UNAVAILABLE
+#endif
+#define OS_SWIFT_UNAVAILABLE(errstring)
+
 
 // These and similar macros come from Availabilty.h on OS X
 // Need a better way to do this long term.
 #define __OSX_AVAILABLE_BUT_DEPRECATED(a,b,c,d) //
 #define __OSX_AVAILABLE_BUT_DEPRECATED_MSG(a,b,c,d,msg) //
+#define __OSX_DEPRECATED(a,b,c) //
+#define __IOS_DEPRECATED(a,b,c) //
+#define __TVOS_DEPRECATED(a,b,c) //
+#define __WATCHOS_DEPRECATED(a,b,c) //
+
+
+#ifdef __OSX_AVAILABLE
+#undef __OSX_AVAILABLE
+#endif
+#define __OSX_AVAILABLE(version)
+
+#ifdef __IOS_AVAILABLE
+#undef __IOS_AVAILABLE
+#endif
+#define __IOS_AVAILABLE(version)
+
+#ifdef __TVOS_AVAILABLE
+#undef __TVOS_AVAILABLE
+#endif
+#define __TVOS_AVAILABLE(version)
+
+#ifdef __WATCHOS_AVAILABLE
+#undef __WATCHOS_AVAILABLE
+#endif
+#define __WATCHOS_AVAILABLE(version)
+
+#ifndef os_likely
+#define os_likely(cond)  (cond)
```

Comment:
this should be `__builtin_expect(!!(cond), 1)`, see the macOS SDK (`<os/base.h>`)